From d7a40ef8e1dd0ca311c7532e0a2e302a7d14ab22 Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Thu, 8 Feb 2024 18:59:09 +0100
Subject: [PATCH] CMakeLists.txt: fixing TMPDIR injections

Resorting to cmake targets and target_link_libraries to fix
TMPDIR injections.

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 CMakeLists.txt                |  4 ---
 ompl_interface/CMakeLists.txt | 48 ++++++++++++++++++++---------------
 2 files changed, 27 insertions(+), 25 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a4ac87f2..cf03117f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,6 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
 endif()
 
 include_directories(ompl_interface/include)
-include_directories(SYSTEM
-                    ${Boost_INCLUDE_DIRS}
-                    ${OMPL_INCLUDE_DIRS})
 
 add_subdirectory(ompl_interface)
 
@@ -34,7 +31,6 @@ install(TARGETS moveit_ompl_interface moveit_ompl_planner_plugin
   INCLUDES DESTINATION include
 )
 ament_export_targets(export_${PROJECT_NAME} HAS_LIBRARY_TARGET)
-ament_export_dependencies(moveit_core ompl)
 
 pluginlib_export_plugin_description_file(moveit_core ompl_interface_plugin_description.xml)
 
diff --git a/ompl_interface/CMakeLists.txt b/ompl_interface/CMakeLists.txt
index ff71c07c0..ca4812a14 100644
--- a/ompl_interface/CMakeLists.txt
+++ b/ompl_interface/CMakeLists.txt
@@ -31,19 +31,23 @@ if(APPLE)
   target_link_directories(${MOVEIT_LIB_NAME} PUBLIC ${OMPL_LIBRARY_DIRS})
 endif()
 
-ament_target_dependencies(${MOVEIT_LIB_NAME}
-  moveit_core
-  moveit_msgs
-  moveit_ros_planning
-  rclcpp
-  pluginlib
-  tf2_eigen
-  tf2_ros
-  OMPL
-  Boost
-)
+target_link_libraries(${MOVEIT_LIB_NAME} PUBLIC
+  moveit_core::moveit_utils
+  moveit_core::moveit_robot_state
+  ${moveit_msgs_TARGETS}
+  moveit_ros_planning::moveit_planning_scene_monitor
+  moveit_ros_planning::moveit_constraint_sampler_manager_loader
+  moveit_core::moveit_planning_interface
+  rclcpp::rclcpp
+  pluginlib::pluginlib
+  tf2_eigen::tf2_eigen
+  tf2_ros::tf2_ros
+  ompl::ompl
+  Boost::boost)
+
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES COMPILE_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES LINK_FLAGS "${OpenMP_CXX_FLAGS}")
+set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 
 add_executable(moveit_generate_state_database scripts/generate_state_database.cpp)
 target_link_libraries(moveit_generate_state_database ${MOVEIT_LIB_NAME})
@@ -52,16 +56,18 @@ set_target_properties(moveit_generate_state_database PROPERTIES OUTPUT_NAME "gen
 
 add_library(moveit_ompl_planner_plugin SHARED src/ompl_planner_manager.cpp)
 set_target_properties(moveit_ompl_planner_plugin PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-ament_target_dependencies(moveit_ompl_planner_plugin
-  moveit_core
-  moveit_ros_planning
-  rclcpp
-  pluginlib
-  tf2_ros
-  OMPL
-  Boost
-)
-target_link_libraries(moveit_ompl_planner_plugin ${MOVEIT_LIB_NAME})
+target_link_libraries(moveit_ompl_planner_plugin PUBLIC
+  moveit_core::moveit_utils
+  moveit_core::moveit_robot_state
+  moveit_ros_planning::moveit_planning_scene_monitor
+  moveit_ros_planning::moveit_constraint_sampler_manager_loader
+  moveit_core::moveit_planning_interface
+  rclcpp::rclcpp
+  pluginlib::pluginlib
+  tf2_ros::tf2_ros
+  ompl::ompl
+  Boost::boost
+  moveit_ompl_interface)
 
 install(TARGETS moveit_generate_state_database
   RUNTIME DESTINATION lib/${PROJECT_NAME}
-- 
2.43.0

